laborator/lab-07/README.md

```diff
@@ -0,0 +1,464 @@
+IOCLA - [https://github.com/systems-cs-pub-ro/iocla](https://github.com/systems-cs-pub-ro/iocla).
+Repository-ul este clonat pe desktop-ul
+mașinii virtuale. Pentru a îl actualiza, folosiți comanda **git pull origin master** din interiorul
+directorului în care se află repository-ul (**~/Desktop/iocla**). Recomandarea este să îl actualizați
+cât mai frecvent, înainte să începeți lucrul, pentru a vă asigura că aveți versiunea cea mai recentă.
+Dacă doriți să descărcați repository-ul în altă locație, folosiți comanda
+**git clone [https://github.com/systems-cs-pub-ro/iocla](https://github.com/systems-cs-pub-ro/iocla) ${target}**.
+Pentru mai multe informații despre folosirea utilitarului **git**, urmați ghidul de la [Git Immersion](https://gitimmersion.com/).
+
+
+### 0. Recapitulare: Șirul lui Fibonacci
+
+Completați fișierul **fibo.asm** din arhivă pentru a realiza un program care afișează primele N numere din șirul lui Fibonacci.
+
+Aveți voie să folosiți doar memorie alocată pe stivă.
+
+### 2. Hello, world!
+
+Deschideți fișierul **hello-world.asm**, asamblați-l și rulați-l. Observați afișarea mesajului *Hello, world*!
+
+Remarcați că:
+- Programul **hello-world.asm** folosește apelul funcției **    puts** (funcție externă modulului curent) pentru a efectua afișarea. Pentru aceasta pune argumentul pe stivă și apelează funcția.
+- Variabila **msg** din programul **hello-world.asm** conține octetul **10**. Acesta simbolizează caracterul *line-feed*, mai cunoscut și sub forma **\n**, folosit pentru a adăuga o linie nouă pe Linux.
+
+Încheierea cu **\n** este, în general, utilă pentru afișarea șirurilor. Funcția **puts** pune automat o linie nouă după șirul afișat, însă aceasta trebuie adăugată explicit în cazul folosirii funcției **printf**. 
+
+### 3. Dezasamblarea unui program scris în C
+
+După cum spuneam, în final, totul ajunge în limbaj de asamblare (ca să fim 100% corecți, totul ajunge cod mașină care are o corespondență destul de bună cu codul asamblare). Adesea ajungem să avem acces doar la codul obiect al unor programe și vrem să inspectăm modul în care arată.
+
+Pentru a observa acest lucru, haideți să compilăm până la codul obiect un program scris în C și apoi să-l dezasamblăm. Este vorba de programul test.c din arhiva de laborator. 
+
+>**NOTE**:<br>
+Pentru a compila un fișier cod sursă C/C++ în linia de comandă, urmați pașii:<br>
+    1. Deschideți un terminal. (shortcut Ctrl+Alt+T)<br>
+    2. Accesați directorul în care aveți codul sursă.<br>
+    3. Folosiți comanda:<br>
+    ```
+    gcc -m32 -o <executabil> <nume-fisier>
+    ```<br>
+    unde <nume-fisier> este numele fișierului iar <executabil> este executabilul rezultat.<br>
+    4. Dacă doriți doar să compilați fișierul (fără să-l link-ați), atunci folosiți comanda <br>
+    ```
+    gcc -m32 -c -o <fisier-obiect> <nume-fisier>
+    ```<br>
+    unde <nume-fisier> este numele fișierului iar <fisier-obiect> este fișierul obiect rezultat.
```

Comment:
Folosirea `<br>`-urilor în general nu este good practice. Poți începe fiecare linie cu `>` pentru a se înțelege că este vorba de aceeași notă.

Observațiile ar trebui să aibă tipul pe aceeași linie cu prima linie de text - `**NOTE**: Pentru a...`.

Acolo unde folosești trei backtick-uri ar fi bine să fie specificat și tipul de sintaxă - aici ar fi bash pentru că sunt comenzi pe care le-am folosi în terminal.